electricitymap/contrib/lib/models/events.py

```diff
@@ -119,11 +154,31 @@ def set_value(
         If correct_negative_with_zero is set to True, negative values will be set to 0 instead of None.
         This method keeps track of values that have been corrected.
         """
-        if correct_negative_with_zero and value is not None and value < 0:
-            value = 0
-            self._corrected_negative_values.add(mode)
+        value = self._correct_negative_value(mode, value, correct_negative_with_zero)
         self.__setattr__(mode, value)
 
+    def add_value(
```

Comment:
Since we are doing OOP, how about bringing some part inside the base Mix class?
I think the part about adding the value can be shared.

Reply:
They already are in the base mix class?
This just overides that so it also corrects for negative values which we don't want to do for storage but I guess we could move it all up and use a flag instead?

Reply:
Indeed, I didn't expended the code so I didn't see it was already in the Mix.
Then you can directly go for my suggestion below